src/NuGet.Clients/NuGet.PackageManagement.UI/Options/ConfigurationFilesControl.xaml

```diff
@@ -0,0 +1,67 @@
+<UserControl x:Class="NuGet.PackageManagement.UI.Options.ConfigurationFilesControl"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:nuget="clr-namespace:NuGet.PackageManagement.UI"
+             xmlns:options="clr-namespace:NuGet.PackageManagement.UI.Options"
+             Background="{DynamicResource {x:Static SystemColors.ControlBrushKey}}"
+             Foreground="{DynamicResource {x:Static SystemColors.ControlTextBrushKey}}"
+             mc:Ignorable="d" 
+             d:DesignHeight="450" d:DesignWidth="800">
+  <UserControl.Resources>
+    <ResourceDictionary>
+      <ResourceDictionary.MergedDictionaries>
+        <options:SharedResources />
+      </ResourceDictionary.MergedDictionaries>
+      <Style TargetType="ListViewItem">
+        <EventSetter Event="MouseDoubleClick" Handler="ListViewItem_MouseDoubleClick"/>
+      </Style>
+    </ResourceDictionary>
+  </UserControl.Resources>
+  <UserControl.CommandBindings>
+    <CommandBinding
+      Command="{x:Static nuget:PackageManagerControlCommands.OpenExternalLink}"
+      Executed="ExecuteOpenExternalLink" />
```

Comment:
I know very little about WPF, but is this command binding, binding to a static class's static field, expected?

Since it's static, I'm concerned that this binding will overwrite whatever static instance is saved in that field, and therefore  whatever has binded (bound?) to it earlier will now execute the wrong command when clicked?

Maybe I'm off-base because of my limited WPF knowledge, but I feel like either:

1. Use a static routed command, but don't set the "executed" handler, it should be set up by the static itself, so it's always the same for everyone (the field should probably be `readonly` too, or a get-only property, to prevent these types of reassigning issues)

or

2. it should bind to a field/property on the viewmodel, not a static.

Reply:
The preferred way to handle this would be number 2 from what @zivkan suggested.

Reply:
We actually do this in 4 other places, example: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/DetailControl.xaml#L24-L26 and it works. We decided to use the same implementation that we currently use in the project, should we change this in all of the places?

Reply:
Yes please. I'd like us to improve the MVVM delineation in PM UI as we do work in it.

Reply:
I spent some time looking at this with @dainizzz in further detail. I better understand why this RoutedCommand is defined here now.

The RoutedCommand is designed to handle an event that is passed up the element tree to somewhere a CommandBinding is defined and then uses the Execute and CanExecute methods to handle the command. This is the way the DetailsControl is set up to use it and it works... however, this adds extra complexity and indirection that isn't necessary.

The [Hyperlink](https://learn.microsoft.com/en-us/dotnet/api/system.windows.documents.hyperlink?view=windowsdesktop-7.0) object has an event [RequestNavigate](https://learn.microsoft.com/en-us/dotnet/api/system.windows.documents.hyperlink.requestnavigateevent?view=windowsdesktop-7.0#system-windows-documents-hyperlink-requestnavigateevent) that is responsible for handling the click of a hyperlink that has a NavigateUri attribute specified. The NavigateUri gets passed in the RequestNavigateEventArgs in its Uri property.

We should be handling this RequestNavigate event on the UI and delegating the responsibility of validating the Uri and launching the external process to the UIUtility.

Doing this allows us to get rid of our dependency on the PackageManagerControlCommands altogether, it removes the need to create a RoutedCommand, it removes the need for us to specify a Command on the Hyperlink and instead we can rely on the built-in event that's designed to handle URI navigation.

And lastly, I asked myself if it's best to put the navigation of to the URI in the View or in the ViewModel and I think the View is the appropriate place. Why? Because if this same ViewModel were to be used for a CommandLine utility or a website, the way the URI would be navigated to would be vastly different. It would be dependent on how the UI was designed as to how it wanted to display the content from that URI, so the responsibility of how to navigate appropriately lies with the View.

Reply:
I show my suggestion in code [here](https://github.com/NuGet/NuGet.Client/pull/5292/files#r1274003977)